usrtools/txgen/app/cmds.c

```diff
@@ -977,3 +983,168 @@ enable_capture(port_info_t *info, uint32_t state)
 {
     txgen_set_capture(info, state);
 }
+
+/**
+ *
+ * start_stop_latency_sampler - Starts or stops latency sampler.
+ *
+ * DESCRIPTION
+ * Starts or stops latency sampler.
+ *
+ * RETURNS: N/A
+ *
+ * SEE ALSO:
+ */
+
+void
+start_stop_latency_sampler(port_info_t *info, uint32_t state)
+{
+    if (state == ENABLE_STATE)
+        txgen_start_latency_sampler(info);
+    else if (state == DISABLE_STATE)
+        txgen_stop_latency_sampler(info);
+}
+
+/**
+ *
+ * txgen_start_latency_sampler - Starts latency sampler.
+ *
+ * DESCRIPTION
+ * Starts latency sampler.
+ *
+ * RETURNS: N/A
+ *
+ * SEE ALSO:
+ */
+
+void
+txgen_start_latency_sampler(port_info_t *info)
+{
+
+    /* Start sampler */
+    if (txgen_tst_port_flags(info, SAMPLING_LATENCIES))
+        return;
+
+    if (info->latsamp_rate == 0 || info->latsamp_type == LATSAMPLER_UNSPEC ||
+        info->latsamp_num_samples == 0) {
+        CNE_ERR("Set proper sampling type, number, rate and outfile!");
+        return;
+    }
+
+    info->latsamp_stats.pkt_counter = 0;
+    info->latsamp_stats.next        = 0;
+    info->latsamp_stats.idx         = 0;
+    info->latsamp_stats.num_samples = info->latsamp_num_samples;
+
+    if (info->pkt.pktSize < (CNE_ETHER_MIN_LEN - ETHER_CRC_LEN) + sizeof(tstamp_t))
+        info->pkt.pktSize += sizeof(tstamp_t);
```

Comment:
info->pkt.pktsize could be much smaller than (CNE_ETHER_MIN_LEN -ETHER_CRC_LEN) + sizeof(tstamp_t), if true we only add sizeof(tstamp_t), may be need to make sure is set to exactly the above length.

Reply:
yeah will fix